WordPress/gutenberg
# [RNMobile] Fix Fixed warning in `clean` script

## What?

This PR fixes the following two problems with scripts in the `react-native-editor` and `react-native-aztec` packages:

- In the `react-native-editor` package, there are scripts that contain a `clean:jest` script that does not exist.
- `clean-watchman` fails in the `react-native-aztec` package if watchman is not installed

For example, if you run the following script, trunk will log errors.

<details><summary>npm run native start:reset</summary>

```bash
$npm run native start:reset

> gutenberg@16.2.0-rc.1 native /home/{username}/projects/gutenberg
> npm run --prefix packages/react-native-editor "start:reset"


> @wordpress/react-native-editor@1.99.0 start:reset /home/{username}/projects/gutenberg/packages/react-native-editor
> npm run clean:runtime && npm run start:quick -- --reset-cache


> @wordpress/react-native-editor@1.99.0 clean:runtime /home/{username}/projects/gutenberg/packages/react-native-editor
> npm run clean:haste; npm run clean:metro; npm run clean:jest; npm run clean:watchman; npm run clean:babel-cache


> @wordpress/react-native-editor@1.99.0 clean:haste /home/{username}/projects/gutenberg/packages/react-native-editor
> rm -rf $TMPDIR/haste-map-metro-*


> @wordpress/react-native-editor@1.99.0 clean:metro /home/{username}/projects/gutenberg/packages/react-native-editor
> rm -rf $TMPDIR/metro-cache/

npm ERR! missing script: clean:jest
npm ERR! 
npm ERR! Did you mean one of these?
npm ERR!     clean:haste
npm ERR!     clean:metro
npm ERR!     clean

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/{username}/.npm/_logs/2023-07-09T10_07_11_648Z-debug.log

> @wordpress/react-native-editor@1.99.0 clean:watchman /home/{username}/projects/gutenberg/packages/react-native-editor
> command -v watchman >/dev/null 2>&1 && watchman watch-del-all; true


> @wordpress/react-native-editor@1.99.0 clean:babel-cache /home/{username}/projects/gutenberg/packages/react-native-editor
> rm -rf ../../node_modules/.cache/babel-loader/*


> @wordpress/react-native-editor@1.99.0 start:quick /home/{username}/projects/gutenberg/packages/react-native-editor
> react-native start "--reset-cache"

                                                      
                        #######                       
                   ################                   
                #########     #########               
            #########             ##########          
        #########        ######        #########      
       ##########################################     
      #####      #####################       #####    
      #####          ##############          #####    
      #####    ###       ######       ###    #####    
      #####    #######            #######    #####    
      #####    ###########    ###########    #####    
      #####    ##########################    #####    
      #####    ##########################    #####    
      #####      ######################     ######    
       ######        #############        #######     
         #########        ####       #########        
              #########          #########            
                  ######### #########                 
                       #########                      
                                                      
                                                      
warning: the transform cache was reset.
                    Welcome to Metro!
              Fast - Scalable - Integrated



To reload the app press "r"
To open developer menu press "d"
```

</details> 

<details><summary>npm run native clean</summary>

```bash
$ npm run native clean

> gutenberg@16.2.0-rc.1 native /home/{username}/projects/gutenberg
> npm run --prefix packages/react-native-editor "clean"


> @wordpress/react-native-editor@1.99.0 clean /home/{username}/projects/gutenberg/packages/react-native-editor
> npm run clean:build-artifacts; npm run clean:aztec; npm run clean:haste; npm run clean:jest; npm run clean:metro; npm run clean:watchman


> @wordpress/react-native-editor@1.99.0 clean:build-artifacts /home/{username}/projects/gutenberg/packages/react-native-editor
> rm -rf ./ios/build && rm -rf ./ios/Pods


> @wordpress/react-native-editor@1.99.0 clean:aztec /home/{username}/projects/gutenberg/packages/react-native-editor
> cd ../react-native-aztec && npm run clean


> @wordpress/react-native-aztec@1.99.0 clean /home/{username}/projects/gutenberg/packages/react-native-aztec
> npm run clean-watchman; npm run clean-node; npm run clean-react; npm run clean-metro; npm run clean-jest;


> @wordpress/react-native-aztec@1.99.0 clean-watchman /home/{username}/projects/gutenberg/packages/react-native-aztec
> command -v watchman >/dev/null 2>&1 && watchman watch-del-all;

npm ERR! code ELIFECYCLE
npm ERR! syscall spawn
npm ERR! file sh
npm ERR! errno ENOENT
npm ERR! @wordpress/react-native-aztec@1.99.0 clean-watchman: `command -v watchman >/dev/null 2>&1 && watchman watch-del-all;`
npm ERR! spawn ENOENT
npm ERR! 
npm ERR! Failed at the @wordpress/react-native-aztec@1.99.0 clean-watchman script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm WARN Local package.json exists, but node_modules missing, did you mean to install?

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/{username}/.npm/_logs/2023-07-09T10_07_43_543Z-debug.log

> @wordpress/react-native-aztec@1.99.0 clean-node /home/{username}/projects/gutenberg/packages/react-native-aztec
> rm -rf node_modules/;


> @wordpress/react-native-aztec@1.99.0 clean-react /home/{username}/projects/gutenberg/packages/react-native-aztec
> rm -rf $TMPDIR/react-*; rm -rf $TMPDIR/react-native-packager-cache-*;


> @wordpress/react-native-aztec@1.99.0 clean-metro /home/{username}/projects/gutenberg/packages/react-native-aztec
> rm -rf $TMPDIR/metro-cache-*; rm -rf $TMPDIR/metro-bundler-cache-*;


> @wordpress/react-native-aztec@1.99.0 clean-jest /home/{username}/projects/gutenberg/packages/react-native-aztec
> rm -rf $TMPDIR/jest_*;


> @wordpress/react-native-editor@1.99.0 clean:haste /home/{username}/projects/gutenberg/packages/react-native-editor
> rm -rf $TMPDIR/haste-map-metro-*

npm ERR! missing script: clean:jest
npm ERR! 
npm ERR! Did you mean one of these?
npm ERR!     clean:haste
npm ERR!     clean:metro
npm ERR!     clean

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/{username}/.npm/_logs/2023-07-09T10_07_44_808Z-debug.log

> @wordpress/react-native-editor@1.99.0 clean:metro /home/{username}/projects/gutenberg/packages/react-native-editor
> rm -rf $TMPDIR/metro-cache/


> @wordpress/react-native-editor@1.99.0 clean:watchman /home/{username}/projects/gutenberg/packages/react-native-editor
> command -v watchman >/dev/null 2>&1 && watchman watch-del-all; true
```

</details> 

## How?

In the `react-native-editor` package, I have removed `clean:jest` from the scripts. The script itself was removed in #51869.

In the `react-native-aztec` package, I added `true` at the end to prevent an error if watchman is not installed.

## Testing Instructions

Confirm that the following two codes do not log errors:

- `npm run native start:reset`
- `npm run native clean`

This change should not affect the application, but I am not familiar with it, so if there is a problem I would be glad to hear about it.
